src/views/Exam/Exam.utils.ts

```diff
@@ -11,13 +11,26 @@ export function getObjectValue<T>(
 export const getRandomQuestions = (
   questions: Question[],
   questionsCount: number,
+  learntQuestions?: LearntQuestions,
 ) => {
-  return shuffle(questions, { copy: true }).slice(0, questionsCount);
+  return shuffle(
+    learntQuestions?.length
+      ? questions.filter(({ id }) => !learntQuestions.includes(id))
```

Comment:
Good idea, I've even done a quick search on the topic, and I found out that it can be taken even further with Sets: https://github.com/anvaka/set-vs-object